DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -628,9 +628,10 @@ struct DataHeader : public BaseHeader {
   using SplitPayloadIndexType = uint32_t;
   using SplitPayloadPartsType = uint32_t;
   using PayloadSizeType = uint64_t;
+  using GlobalIDType = uint64_t;
```

Comment:
@mkrzewic  actually, uint32_t would be enough since the orbit is anyway limited by 32 bits.

Reply:
I think it would be nice if it could be unique globally, for e.g. processing entire data sets. maybe it can also contain some starting point (run number, run start timestamp or something). We have been talking about it forever but no agreement yet...

Reply:
I don't think anybody will maintain a global TF ID trough data dating, moreover that the TF length might not be fixed. More universal identification would be provided by the run_number (which allows to access orbit/bc at the start of run) + current orbit/bc, but I am not sure what is the role of this info in the transient data header.  
The TF ID with ID unique within the run would be useful for synchronization, but what would you do with extra info?

Reply:
I agree, the name "GlobalIDType" seems misleading. TimeFrameID is more clear and to the point. AFAIK, it's only unique within the data set of a single run. Anyone requiring absolute time of the data must look somewhere else for it. Aim of having TFID in the header is to match all data peaces during the dataflow and processing and the final CTF merger.

Reply:
Well. Having the time is a requirement from Data Processing, so you will have to provide it in any case, given it's STFB which interfaces with the trigger. Any reason for not using the time as timeframe id? As I understand it, it is provided by the readout.

Reply:
What STFB knows is the following: the TFID, and the first orbit of the TF. 
CTP to my knowledge does not distribute time. How to work out the absolute "time" from this has not been discussed in this context. Should the processing requires more information this needs to be discussed with all relevant subsystems.

Reply:
So can we use the first orbit of the TF for the ID, which is time-like beside an offset (correct?)? What data processing needs is a monotonically increasing quantity where equal distances in the value correspond to equal distances in time.

Reply:
We can then discuss who is responsible to deliver the offset for the run and the scale in a separate meeting. I would still find it useful to have it in the header so that we do not need two extra parameter to provided by control / some database to query CCDB.

Reply:
Duration could also be (last orbit - first orbit) to start with.

Reply:
@ktf then what is wrong with simple TF counter? Or think about the CCDB access on the framework level?

Reply:
ok, got @ktf messages after asking about the CCDB. But you have the time in the DataProcessingHeader, can't you use that

Reply:
yes, i can use that, but:
1) why not removing the need for a separate header?
2) who will fill the DataProcessingHeader?

Reply:
if you want to merge headers, it is fine, but given that the timestamp for the CCDB access will be needed also during async. processing, the absolute time stamp should be extractable  from the TF data themselves and not expected to be provided by additional source via transient headers.
In principle, the info of the orbit/bc at start_of_run + absolute its time combined with the 1st orbit of the TF is enough to build the abs. timestamp of the TF. The run start info is supposed to come from the CTP (and must be stored in the CCDB). At the moment I have in the GRP object only the abs. time of (simulated) run start, can add also the orbit/bc.

Reply:
@shahor02 I agree for the async phase, it's easy to extract a time / duration from the file the time and use it in DPL. The problem for the sync phase is that I do not have any way to extract such information if the STFB does not provide it to me in a readily available manner. Clearly we do not want to scan the whole timeframe payload, nor we want to query a DB just to extract a time to start the processing.

Reply:
I agree with what @shahor02 proposes. However, @kft, I cannot provide the absolute timing information. Component that issues start of the run (possibly restarting the global orbit counter) must make this information available. Iguess this it CTP or DCS. Until STFBs see any data it could be milliseconds, and the absolute time could differ across all FLPs.